modules/apps/data-engine/data-engine-service/src/main/java/com/liferay/data/engine/internal/data/engine/content/type/DataEngineIntrinsicObjectDataDefinitionContentType.java

```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.data.engine.internal.data.engine.content.type;
+
+import com.liferay.data.engine.content.type.DataDefinitionContentType;
+import com.liferay.data.engine.intrinsic.object.DataEngineIntrinsicObject;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.security.permission.PermissionChecker;
+import com.liferay.portal.kernel.util.Portal;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Jeyvison Nascimento
+ */
+@Component(
+	immediate = true, property = "content.type=intrinsic-object",
+	service = DataDefinitionContentType.class
+)
+public class DataEngineIntrinsicObjectDataDefinitionContentType
+	implements DataDefinitionContentType {
+
+	@Override
+	public long getClassNameId() {
+		return _portal.getClassNameId(DataEngineIntrinsicObject.class);
+	}
+
+	@Override
+	public String getContentType() {
+		return "intrinsic-object";
+	}
+
+	@Override
+	public String getPortletResourceName() {
+		return "com.liferay.app.builder";
```

Comment:
is this a content type that should belong to app builder or data engine? if it's just a app builder thing, should be moved to the app builder module.